security/pkg/nodeagent/caclient/client.go

```diff
@@ -44,10 +46,13 @@ type configMap interface {
 }
 
 // NewCAClient create an CA client.
-func NewCAClient(endpoint, CAProviderName string, tlsFlag bool) (caClientInterface.Client, error) {
+func NewCAClient(endpoint, CAProviderName string, tlsFlag bool, vaultAddr, vaultRole,
+	vaultAuthPath, vaultSignCsrPath string) (caClientInterface.Client, error) {
```

Comment:
don't pass vault specific params in this func, better to get those env variable in vault client.

Reply:
Putting the CA provider inside vault client has pros and cons.
- Pros: less perimeters. 
- Cons: CA provider is not exposed at Node Agent. The code is difficult to understand because how the CA provider is configured is hidden somewhere.

Reply:
those variables are CA provider specific, since we already passed ```CAProviderName``` as input param, setting those variables should be pushed to lower layer(each plugin). let's revisit this after e2e works.